lib/presentation/screen/booking/widgets/calendar_widget.dart

```diff
@@ -0,0 +1,383 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:table_calendar/table_calendar.dart';
+
+import '../../../design_system/colors/app_color.dart';
+import '../../../design_system/typography/app_typography.dart';
+
+class CalendarWidget extends StatefulWidget {
+  final Set<DateTime> unavailableDates;
+
+  const CalendarWidget({super.key, this.unavailableDates = const {}});
```

Comment:
Passing unavailableDates has no any effect, because you use fake data `fakeUnavaliableDates` 

please remove fake data then use real state